lib/appc/manifest.go

```diff
@@ -0,0 +1,134 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appc
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/appc/spec/aci"
+	"github.com/appc/spec/schema"
+	"github.com/appc/spec/schema/types"
+)
+
+var (
+	// ErrNotFound is returned when acbuild is asked to remove an element from a
+	// list and the element is not present in the list
+	ErrNotFound = fmt.Errorf("element to be removed does not exist in this ACI")
+)
+
+// Manifest is a struct with an open handle to a manifest that it can manipulate
+type Manifest struct {
+	aciPath      string
+	manifest     *schema.ImageManifest
+	manifestFile *os.File
+}
+
+// LoadManifest will read in the manifest from an untarred ACI on disk at
+// location aciPath, and return a new Manifest struct to manipulate it.
+func LoadManifest(aciPath string) (*Manifest, error) {
+	manFile, err := os.OpenFile(path.Join(aciPath, aci.ManifestFile), os.O_RDWR, 0644)
+	if err != nil {
+		return nil, err
+	}
+
+	manblob, err := ioutil.ReadAll(manFile)
+	if err != nil {
+		manFile.Close()
+		return nil, err
+	}
+
+	man := &schema.ImageManifest{}
+	err = man.UnmarshalJSON(manblob)
+	if err != nil {
+		manFile.Close()
+		return nil, err
+	}
+
+	return &Manifest{aciPath, man, manFile}, nil
+}
+
+// Close will close the Manifest struct's open handle to the manifest file
+func (m *Manifest) Close() error {
+	err := m.manifestFile.Close()
+	if err != nil {
+		return err
+	}
+	m.manifestFile = nil
+	return nil
+}
+
+// save commits changes to m's manifest to disk
+func (m *Manifest) save() error {
+	blob, err := m.manifest.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	_, err = m.manifestFile.Seek(0, os.SEEK_SET)
+	if err != nil {
+		return err
+	}
+
+	err = m.manifestFile.Truncate(0)
+	if err != nil {
+		return err
+	}
+
+	_, err = m.manifestFile.Write(blob)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Get returns the manifest currently being manipulated
+func (m *Manifest) Get() *schema.ImageManifest {
+	return m.manifest
+}
+
+// Print will print out the current manifest to stdout, optionally inserting
+// whitespace to improve readability
+func (m *Manifest) Print(prettyPrint, printConfig bool) error {
+	if printConfig {
+		return fmt.Errorf("can't print config, appc has no image configs")
+	}
+	var manblob []byte
+	var err error
+	if prettyPrint {
+		manblob, err = json.MarshalIndent(m.manifest, "", "    ")
+	} else {
+		manblob, err = m.manifest.MarshalJSON()
+	}
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(manblob))
```

Comment:
Sorry, it was a mechanical comment, in the context of this function it does make sense, you are right.

Nevertheless, since this lives in `lib`, I'd rather pass an `io.Writer` explicitely and use `fmt.Fprintln` here.

What do you think?